core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -692,6 +683,25 @@ protected void closeFailed(Throwable cause) {
         private void logStopFailure(Throwable cause) {
             logger.warn("Failed to stop a server: {}", cause.getMessage(), cause);
         }
+
+        private void shutdownExecutor(ScheduledExecutorService executor) {
+            if (executor instanceof UnstoppableScheduledExecutorService) {
+                executor = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
+            }
+
+            try {
+                executor.shutdown();
+                while (!executor.isTerminated()) {
+                    try {
+                        executor.awaitTermination(1, TimeUnit.DAYS);
+                    } catch (InterruptedException ignore) {
```

Comment:
Could we call `shutdown()` for all blocking task executors first and then wait for their termination, so that the shutdown process takes less time?